runtime/lua/vim/diagnostic.lua

```diff
@@ -643,6 +619,36 @@ end
 ---                * priority: (number, default 10) Base priority to use for signs. When
 ---                {severity_sort} is used, the priority of a sign is adjusted based on
 ---                its severity. Otherwise, all signs use the same priority.
+---                * sign_text: (table) Table of vim.diagnostic.severity symbol pairs.
+---                Text values in this table take priority over existing sign definitions
+---                but will not be prioritized over a sign_text field in the user data of an individual
+---                diagnostic. Individual diagnostic's sign_text can be set within a function
+---                passed to the filter parameter.
+---                * filter: (function) A function which will be run prior to determining the signs
+---                This function takes an individual diagnostic as a parameter and can be used to filter
+---                or otherwise customize how signs will appear by setting the sign_text, sign_hl_group or
+---                priority fields for the diagnostic. If a diagnostic is not returned, no sign will be created
```

Comment:
```suggestion
---                * filter: (function) A function which takes an individual diagnostic as its
---                single argument and optionally returns a diagnostic. It can be used to filter or
---                otherwise customize how signs will appear by setting the "sign_text",
---                "sign_hl_group", or "sign_priority" fields in the diagnostic's "user_data" table.
---                If nil is returned, no sign is created for the given diagnostic.
```